app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(in_func: Callable) -> Callable:
+    result_dict = {}
+
+    def inner(*args, **kwargs) -> Callable:
+        kwargs_tuple = tuple(kwargs.items())
+        if (args, kwargs_tuple) in result_dict:
```

Comment:
it's better to use a variable where you store arguments. For example `storage_key = (args, tuple(kwargs.items()))`. And then use it in your  code.

Reply:
done!

Reply:
ok